src/app/profile/validators/email.validator.ts

```diff
@@ -0,0 +1,11 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from "@angular/forms";
+
+export function customEmailValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+
+      return !control.value.includes('@') ? { invalidEmail: { value: control.value } } : null;
+      
+    };
+}
+
```

Comment:
COMMENT: This was hard. I had to eventually find some sample code. I dont understand how there are two nested return statements and how that is working. I feel like the first one is for the ValidatorFn to work and the second one is obviously the validation i'm trying to do, but it looks weird to me.

Reply:
so what you're doing is essentially returning a function that returns another function.  The angular designers took a functional programming approach to their design.  That inner function fires when it comes time to validate the control.  They probably execute the outer function at the time the form is built.

Reply:
I think it would look a little easier to read like this:

```
export function customEmailValidator(): ValidatorFn {
    return (control: AbstractControl): ValidationErrors | null => {

      const errorToAddToControl = { invalidEmail: { value: control.value } };
      
      return !control.value.includes('@') 
        ? errorToAddToControl 
        : null;
    };
}
```